crates/re_types/src/datatypes/tensor_data_ext.rs

```diff
@@ -226,6 +256,46 @@ impl TensorData {
         }
     }
 
+    pub fn get_yuv422_pixel(&self, x: u64, y: u64) -> Option<[TensorElement; 3]> {
+        let TensorBuffer::Yuv422(buf) = &self.buffer else {
+            return None;
+        };
+
+        match self.image_height_width_channels() {
+            Some([_, w, _]) => {
+                // given an x and y coordinate, get the offset into the yuv422 buffer
+                let index = ((y * w + x) * 2) as usize;
+                let (luma, u, v) = if x % 2 == 0 {
+                    (buf[index], buf[index + 1], buf[index + 3])
+                } else {
+                    (buf[index], buf[index - 1], buf[index + 1])
+                };
+
+                let (r, g, b) = Self::yuv_to_rgb(luma, u, v);
+
+                Some([
+                    TensorElement::U8(r),
+                    TensorElement::U8(g),
+                    TensorElement::U8(b),
+                ])
+            }
+            _ => None,
+        }
+    }
+
+    /// Convert YUV to RGB
+    fn yuv_to_rgb(y: u8, u: u8, v: u8) -> (u8, u8, u8) {
```

Comment:
The `get_nv12_pixel` function should probably also use this

Reply:
that'd be nice yeah :)
I also noticed the same pattern in the decode shader where we could add a similar utility method for the -16/-128 part

also, we should have the same commentry as in `decoding.wgsl` and refer to each other code blocks. We can't avoid this duplication accross languages, but it would be good to have it noted down